spec/models/group_privacy_spec.rb

```diff
@@ -0,0 +1,126 @@
+require 'rails_helper'
+describe Group do
+  let(:group) { create(:group) }
+
+  describe "group_privacy" do
+    it 'gets values for "open" correctly' do
+      group.is_visible_to_public = true
+      group.discussion_privacy_options = 'public_only'
+      expect(group.group_privacy).to eq 'open'
+    end
+
+    it 'gets values for "closed" correctly' do
+      group.is_visible_to_public = true
+      group.discussion_privacy_options = 'public_or_private'
+      expect(group.group_privacy).to eq 'closed'
+      group.discussion_privacy_options = 'private_only'
+      expect(group.group_privacy).to eq 'closed'
```

Comment:
Hm, I've been thinking that we may want to write a custom validator for group privacy, to better ensure people can't submit weird stuff like public groups with only private threads. It'd be nice to see a `expect(group.valid?).to eq true` for these, along with maybe one or two which assert invalidity.